packages/gatsby/src/redux/run-sift.js

```diff
@@ -15,7 +15,14 @@ const prepareQueryArgs = (filterFields = {}) =>
   Object.keys(filterFields).reduce((acc, key) => {
     const value = filterFields[key]
     if (_.isPlainObject(value)) {
-      acc[key === `elemMatch` ? `$elemMatch` : key] = prepareQueryArgs(value)
+      switch (key) {
+        case `not`:
+        case `elemMatch`:
+          acc[`$${key}`] = prepareQueryArgs(value)
```

Comment:
We need to do something similar [here](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/src/db/loki/nodes-query.js#L50) as well. (For context: Gatsby supports a second, not-very-advertised backend that uses `LokiJS`)